contracts/utils/ExchangeRate.sol

```diff
@@ -1,26 +1,36 @@
-//SPDX-License-Identifier: Unlicense                                                                                                                                         [0/0]
+//SPDX-License-Identifier: Unlicense
 pragma solidity ^0.6.12;
 
 import "@openzeppelin/contracts-ethereum-package/contracts/math/SafeMath.sol";
-/* exchangeRate = principal + surplusInterest / bTokenTotalSupply */
 
+/* @title exchange rate calculation */
+/* @notice from bTokens to aTokens and vice versa */
+/* @dev the equation is exchangeRate = principal + surplusInterest / bTokenTotalSupply */
 contract ExchangeRate {
 
-    // calculates token from btoken amount
+    // @notice returns tokens from btokens, pure function
+    /* @param _principal amount in underlying asset deposited into the contract  */
+    /* @param _surplus the interests earned that belong to the depositorReserve */
+    /* @param _btokens the depositor's btokens amount to convert to underlying tokens */
+    /* @return amount in underlying tokens to return to depositor */
     function btokenToToken(uint256 _principal,
                            uint256 _surplus,
                            uint256 _tokenSupply,
-                           uint256 btokens
+                           uint256 _btokens
                            ) pure public returns (uint256){
-        return SafeMath.div(SafeMath.mul(btokens, SafeMath.add(_surplus, _principal)), _tokenSupply);
+        return SafeMath.div(SafeMath.mul(_btokens, SafeMath.add(_surplus, _principal)), _tokenSupply);
     }
 
-    // calculates btokens amount from tokens
+    // @notice returns btokens from tokens, pure function
```

Comment:
One of these exchange rates can be simplified, at least cognitively: eg: if btoken exchange rate for atoken = `X`, then atoken exchange rate for btoken will be `1/X`.

It seems however that doing it the way you have it is cheaper as it will require one extra division operation....

Is the cost neglible, if so, the `1/X` calculation is pretty best, as it makes one function depend on the other can if their is a bug in the calculation then only one function needs to be updated.

Whatever do you think?

Reply:
could you give an example of what the calculation would look like then? 

I'm rather wary about the cost, also yet another devision operation- we don't have the strong grasp on what produces the error yet. Adding additional operations may make the matter worse.